vpr/src/route/router_lookahead.h

```diff
@@ -25,11 +57,29 @@ class ClassicLookahead : public RouterLookahead {
 class MapLookahead : public RouterLookahead {
   protected:
     float get_expected_cost(int node, int target_node, const t_conn_cost_params& params, float R_upstream) const override;
+    void compute(const std::vector<t_segment_inf>& segment_inf) override;
+    void read(const std::string& file) override {
+        (void)file;
+        VPR_THROW(VPR_ERROR_ROUTE, "MapLookahead::read unimplemented");
+    }
+    void write(const std::string& file) const override {
+        (void)file;
+        VPR_THROW(VPR_ERROR_ROUTE, "MapLookahead::write unimplemented");
+    }
 };
 
 class NoOpLookahead : public RouterLookahead {
   protected:
     float get_expected_cost(int node, int target_node, const t_conn_cost_params& params, float R_upstream) const override;
+    void compute(const std::vector<t_segment_inf>& segment_inf) override {
+        (void)segment_inf;
+    }
+    void read(const std::string& file) override {
+        (void)file;
```

Comment:
Probably should throw as well (not clear what asking a no-op lookahead to read/write should do.